### ISSUE TITLE
💅 Object.assign on Accumulators Shouldn't Be Allowed

### ISSUE THREAD

**Issue by mssngr** [biomejs/biome#5277]
### Environment information

```bash
I am using Biome strictly in the IDE, so this n/a for me.
```

### Rule name

no-accumulating-spread

### Playground link

https://biomejs.dev/playground/?code=YwBvAG4AcwB0ACAAYQByAHIAYQB5AE8AZgBPAGIAagBzACAAPQAgAFsAewAKACAAIABoAGUAbABsAG8AOgAgACcAdwBvAHIAbABkACEAJwAKAH0AXQAKAAoAYQByAHIAYQB5AE8AZgBPAGIAagBzAC4AcgBlAGQAdQBjAGUAKAAoAGEAYwBjAHUAbQAsACAAbwBiAGoAKQAgAD0APgAgAE8AYgBqAGUAYwB0AC4AYQBzAHMAaQBnAG4AKABhAGMAYwB1AG0ALAAgAG8AYgBqACkALAAgAHsAfQApAA%3D%3D

### Expected result

The [No Accumulating Spread](https://biomejs.dev/linter/rules/no-accumulating-spread/) rule prevents the `O(n^2)` problem by not allowing the use of object spreader syntax on accumulators. However, object spreader syntax is just syntactical sugar for `Object.assign`. And no error is shown for utilizing `Object.assign` on an accumulator, even though it faces the same `O(n^2)` problem.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by emilyinure** [comment]
Can I take this one?

**Comment by emilyinure** [comment]
Is there a standardized way to pull the string value out of a JsIdentifierExpression aside from using .to_string?

**Comment by mattfysh** [comment]
wait, this shouldn't have gone through - the reason for no spread syntax is because you're creating a new object, eg. `{ ...a, ...b }` is `Object.assign({}, a, b)` and you can see there the first parameter is a new object created in the accumulator, assigned props from a + b, then returned for the next iterator

but if you're accumulating on the memo'd object, you're not creating a new object - you're mutating the in-memory memo similar to the recommended `arr.push()` suggestion for reduced arrays

this should most likely be reverted, or at least not throw if you Object.assign without creating a new object

**Comment by mattfysh** [comment]
Created new issue to resolve: https://github.com/biomejs/biome/issues/6675
